cloud/ec2_util.go

```diff
@@ -868,8 +896,19 @@ func disassociateIPAddressForHost(ctx context.Context, c AWSClient, h *host.Host
 		return nil
 	}
 
-	_, err := c.DisassociateAddress(ctx, &ec2.DisassociateAddressInput{
+	flags, err := evergreen.GetServiceFlags(ctx)
+	if err != nil {
+		return errors.Wrap(err, "getting service flags")
+	}
+	if flags.ElasticIPsDisabled {
+		return errors.Errorf("elastic IP features are disabled, host will not disassociate elastic IP '%s'", h.IPAssociationID)
```

Comment:
If we return error here, doesn't that propagate up to a log? (do we want to maybe no-op here?)